packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -116,8 +120,10 @@ export function FlatTermSelector( { slug, __nextHasNoMarginBottom } ) {
 					: false,
 				taxonomy: _taxonomy,
 				termIds: _termIds,
-				terms: _termIds?.length
-					? getEntityRecords( 'taxonomy', slug, query )
+				terms: _terms
+					? _termIds.map( ( termId ) =>
+							_terms.find( ( term ) => term.id === termId )
+					  )
```

Comment:
We might want to use a more efficient way of sorting here, i.e. via [`Array.prototype.sort`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort), and a suitable comparison function.

OTOH, using `map()` is pretty intuitive here. Also, `sort()` sorts in-place, i.e. retains referential equality, which isn't really desirable here.